includes/wikia/services/ExternalServicesQueryService.class.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Class for querying Wikia Microservices provided by Consul
+ * Using Wikia's Http class and ConsulService
+ */
+class ExternalServicesQueryService extends Service {
+
+	/**
+	 * @var $consulService ConsulService
+	 */
+	protected $consulService;
+	protected $httpOptions = [];
+
+	const URL_SCHEMA = "http://{host}:{port}/{query_path}";
+	const METHOD_GET = 'GET';
+	const METHOD_POST = 'POST';
+
+	public function __construct( ConsulService $consulService ) {
+		$this->consulService = $consulService;
+		return $this;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getHttpOptions()	{
+		return $this->httpOptions;
+	}
+
+	/**
+	 * Options for Http class
+	 * @param array $httpOptions
+	 * @return $this
+	 */
+	public function setHttpOptions( $httpOptions )	{
+		$this->httpOptions = $httpOptions;
+		return $this;
+	}
+
+	protected function getUrl( $queryPath ) {
+		$target = $this->consulService->resolve();
```

Comment:
Will we get errors below if `$target` is false? Should we check that first and exit early? Same below in the caller `$this->request()`.